plugins/cloud-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeServiceImpl.java

```diff
@@ -90,6 +96,38 @@ public InstanceList run() throws Exception {
         return instances;
     }
 
+    @Override
+    public String networkAddress(String gceName) throws IOException {
+        String urlMetadataNetwork = GCE_METADATA_URL + "/" + gceName;
+        logger.debug("get network information for [{}]", gceName);
+        URL url = new URL(urlMetadataNetwork);
+        HttpHeaders headers;
+        try {
+            // hack around code messiness in GCE code
+            // TODO: get this fixed
+            headers = AccessController.doPrivileged(new PrivilegedExceptionAction<HttpHeaders>() {
+                @Override
+                public HttpHeaders run() throws IOException {
+                    return new HttpHeaders();
+                }
+            });
+
+            // This is needed to query meta data: https://cloud.google.com/compute/docs/metadata
+            headers.put("Metadata-Flavor", "Google");
+            HttpResponse response;
+            response = getGceHttpTransport().createRequestFactory()
+                    .buildGetRequest(new GenericUrl(url))
+                    .setHeaders(headers)
+                    .execute();
+            String ip = response.parseAsString();
+            logger.debug("ip found [{}]", ip);
+            return ip;
+        } catch (Exception e) {
+            logger.warn("unable to create GCE Http Client", e);
```

Comment:
@dadoonet I think the log message can still say "Failed to fetch metadata from google" ? more than just client creation can go wrong here..